scripts/ccpp_datafile.py

```diff
@@ -462,29 +490,67 @@ def _retrieve_variable_list(table, suite_name, intent_type=None):
-                for var in gvars:
+    if excl_prot or (intent_type == "host"):
+        host_dict = _find_var_dictionary(table, dict_type="host")
+        if host_dict is not None:
+            hvars = host_dict.find("variables")
+            if hvars is not None:
+                for var in hvars:
                     vname = var.get("name")
-                    vintent = var.get("intent")
-                    protected = _is_variable_protected(table, vname, group_dict)
-                    if (not protected) and (vintent in allowed_intents):
-                        var_set.add(vname)
+                    if excl_prot:
+                        exclude = _is_variable_protected(table, vname,
+                                                         host_dict)
+                    else:
+                        exclude = False
+                    # end if
+                    if intent_type == "host":
+                        if not exclude:
+                            # Add to host variable set
+                            var_set.add(vname)
+                        # end if
+                    else:
+                        if exclude:
+                            # Add to list of protected variables
+                            excl_vars.append(vname)
+                        # end if
                     # end if
                 # end for
             # end if
         # end if
-    # end for
+    # end if
+    if intent_type != "host":
+        group_names = _retrieve_suite_group_names(table, suite_name)
+        for group in group_names:
+            cl_name = group + "_call_list"
+            group_dict = _find_var_dictionary(table, dict_name=cl_name,
+                                              dict_type="group_call_list")
+            if group_dict is not None:
+                gvars = group_dict.find("variables")
+                if gvars is not None:
+                    for var in gvars:
+                        vname = var.get("name")
+                        vintent = var.get("intent")
+                        if excl_prot:
+                            exclude = vname in excl_vars
+                            if not exclude:
+                                exclude = _is_variable_protected(table, vname,
+                                                                 group_dict)
+                            # end if
+                        else:
+                            exclude = False
+                        # end if
+                        if (vintent in allowed_intents) and (not exclude):
+                            var_set.add(vname)
+                        # end if
+                    # end for
+                # end if
+            # end if
+        # end for
+    # end if
     return sorted(var_set)
 
 ###############################################################################
-def datatable_report(datatable, action, sep):
+def datatable_report(datatable, action, sep, excl_prot=False):
```

Comment:
Not a big thing, but `exclude_protected` isn't that much longer than `excl_prot` and much easier to read ...